src/shogun/machine/Machine.cpp

```diff
@@ -66,6 +66,23 @@ bool CMachine::train(CFeatures* data)
 	return result;
 }
 
+void CMachine::fit(CFeatures* features)
+{
+	REQUIRE(train(features), "Failed to fit machine %s\n", get_name());
```

Comment:
maybe we should throw an exception rather than this boolean stuff?

Reply:
require failed -> sg_io->msg(MSG_ERROR, "blabla") -> https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/io/SGIO.cpp#L125

Reply:
Sure but I mean, shouldn't the train method itself just throw an exception?
We don't get any context information here. If train would throw an exception, we could catch it here and then say "training failed for reason X". No?

Reply:
i'm going to just throw exceptions in `fit`. but for now train returns boolean, i keep this so that it is consistent that fit always throw exceptions when failure.